Tools/clinic/clinic.py

```diff
@@ -5033,26 +5033,27 @@ def bad_node(self, node):
         key = f"{parameter_name}_as_{c_name}" if c_name else parameter_name
         self.function.parameters[key] = p
 
-    KwargDict = dict[str | None, Any]
+    KwargDict = dict[str, Any]
 
     @staticmethod
     def parse_converter(annotation: ast.expr | None) -> tuple[str, bool, KwargDict]:
+        msg = "Annotations must be either a name, a function call, or a string."
         match annotation:
             case ast.Constant(value=str() as value):
                 return value, True, {}
             case ast.Name(name):
                 return name, False, {}
             case ast.Call(func=ast.Name(name)):
                 symbols = globals()
-                kwargs = {
-                    node.arg: eval_ast_expr(node.value, symbols)
-                    for node in annotation.keywords
-                }
+                kwargs: dict[str, Any] = {}
+                for node in annotation.keywords:
+                    if not isinstance(node.arg, str):
+                        fail(msg)
```

Comment:
I would give a different message here. If we get here, it's because someone has done something like:

```python
x: int(**{'x': 'foo'})
```

And that _is_ a function call annotation, so the message "Annotations must be either a name, a function call or a string" doesn't make any sense here.

How about failing with the error "Cannot use a kwarg splat in a function-call annotation" here?

Reply:
The existing message is accurate in all the places where it's currently emitted. This is a different code path, so I don't see any harm in giving a specific and precise error message here?

Reply:
This is the patch I had locally that I was working on, FWIW:

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -5044,10 +5044,11 @@ def parse_converter(annotation: ast.expr | None) -> tuple[str, bool, KwargDict]:
                 return name, False, {}
             case ast.Call(func=ast.Name(name)):
                 symbols = globals()
-                kwargs = {
-                    node.arg: eval_ast_expr(node.value, symbols)
-                    for node in annotation.keywords
-                }
+                kwargs: dict[str, Any] = {}
+                for node in annotation.keywords:
+                    if node.arg is None:
+                        fail("Cannot use a kwarg splat in a function-call annotation")
+                    kwargs[node.arg] = eval_ast_expr(node.value, symbols)
```

Reply:
See 288d8ca4f47e1e0e8a108413e54687230d78b115

Reply:
I'm happy to swap my suggestion with _"Cannot use a kwarg splat in a function-call annotation"_ :)